dotnet-authserver/src/TeacherIdentity.AuthServer/AuthenticationState.cs

```diff
@@ -394,6 +394,48 @@ public void OnTrnLookupCompletedAndUserRegistered(User user)
         TrnLookupStatus = user.TrnLookupStatus;
     }
 
+    public void OnUserRegistered(User user)
+    {
+        ThrowOnInvalidAuthenticationMilestone(AuthenticationMilestone.EmailVerified);
+
+        if (EmailAddress is null)
+        {
+            throw new InvalidOperationException($"{nameof(EmailAddress)} is not known.");
+        }
+
+        if (!EmailAddressVerified)
+        {
+            throw new InvalidOperationException($"Email has not been verified.");
+        }
+
+        if (MobileNumber is null)
```

Comment:
The only place where we call this function, we assert MobileNumber, Email, etc. to be true, so I'm not sure if this is necessary, but was following the pattern above

Reply:
Actually just realised that we need to normalise the mobile number, so shouldn't have these assertions